config/tls.go

```diff
@@ -27,19 +27,16 @@ func (c *Config) setTLSConfig() error {
 			return fmt.Errorf("Error reading certificate/key pair: %w", err)
 		}
 
-		var cat tls.ClientAuthType = tls.RequireAndVerifyClientCert
-		if c.AllowUnauthenticatedReads {
-			// This allows us to handle some requests without a valid client
-			// certificate, but then we need to explicitly check for verified
-			// certs on requests that we require auth for.
-			// See server.checkGRPCClientCert and httpCache.hasValidClientCert.
-			cat = tls.VerifyClientCertIfGiven
-		}
-
 		c.TLSConfig = &tls.Config{
 			Certificates: []tls.Certificate{readCert},
 			ClientCAs:    caCertPool,
-			ClientAuth:   cat,
+
+			// This allows us to handle some requests without a valid client
+			// certificate (like the grpc health check service), but then we
+			// need to explicitly check for verified certs on requests that
+			// we require auth for.
+			// See server.checkGRPCClientCert and httpCache.hasValidClientCert.
+			ClientAuth: tls.VerifyClientCertIfGiven,
```

Comment:
This change now exposes metrics (if enabled) and the /status http endpoint to unauthenticated clients. I need to think this through a bit before landing.